documentation/docs/instructor.md

```diff
@@ -1,82 +1,99 @@
+
+`Reminder: The format of the CSV file used by Blackboard Learn may have additional fields; however, Cloud VCL only requires (Last Name, First Name, Username)`.  
+
+## Creating an Environment Definition
+The environment definition is a virtual environment where assignments will be created through defining VMs. View the Environment Definition List page by selecting the `Environment Definitions` tab found in the header. Create an environment definition by clicking `Create Environment Definition` in the top right corner and providing a name. Click on a environment definition title to view any corresponding defined VMs. Environment definitions can be edited or deleted through the Environment Definition List page. 
+
+`You can create multiple Environment Definitions for one Assignment, which can hold one or more VM Definitions.`
+
+## Creating a VM Definition
+By clicking on an environment definition title from the Environment Definition List page, you can view, create, edit, or delete  the Virtual Machine Definitions that have been created. Create a new VM by clicking 'Create VM Definition` in the top right corner. These VM Definitions will be used to load virtual machines into the environment when it's launched through an assignment. Find more details about specific fields in the creation of a VM below:  
+
+### Image 
+Choose the type of image you wish to use: CentOS 7.3, Ubuntu Server 16.04.2, Windows Server 2012 R2
+### Flavor
+Select the type of flavor: 01c-01m-10d reads 1 VCPU, 1 RAM GB, and 10 GB Total/Root Disk.
+### Shell Script
+The instance will run the script you provide here once it's created. Provide the appropriate shebang respective to the image type:
+
+Windows:
+```
+#ps1_sysnative 
+```
+
+Linux:
+```
+#!/bin/sh
+```
+
+### Install Packages (Linux Only)
+Install packages by inserting one package name per line. You can search for package names here: http://packages.ubuntu.com/.
+
+### Update Package Lists on Boot (Linux Only) 
+By selecting yes, it will update the list of available packages and their versions, but does not install or upgrade any packages.
+
+### Upgrade All Packages on Boot (Linux Only)
+By selecting yes, it will upgrade newer versions of the packages you have. 
+
+### Perform Reboot if Required After Package Install/Upgrade (Linux Only)
+By selecting yes, it will perform a reboot if packages are installed and upgraded.
+
+### Timezone (Linux Only)
+Set a different system timezone. 
+
 Example:
 ```
-American/New_York
+America/New_York
 ```
 
-## Hostname (Linux Only)
-Set the system hostname to be different than default value.
+### Hostname (Linux Only)
+Set a hostname that will replace the hostname generated by default.
+
+### Default User Password (Linux Only)
+Sets the password for default user (username is OS-dependant). Leave blank to disable.
+
+### Default User Public Key (Linux Only)
+SSH public key for default user (username is OS-dependant).
+
+Example:
+```
+ssh-rsa AAAB... user@host
```

Comment:
It's not a full example, give examples of multiple types of public keys. Full ones, the dots are obviously standing for majority of characters.